torch/_refs/__init__.py

```diff
@@ -923,7 +923,8 @@ def sum(
             dtype = torch.int64
         else:
             dtype = a.dtype
-
+    if dim == () or dim == []:
+        dim = None
```

Comment:
the need to test against both empty tuple and empty list seems smelly. Is the reason a false-ish test not used because dim could be zero? It seems better to introduce some parsing logic that will let us uniformly handle int list arguments (and it will be easier to maintain API compatibility with real torch argument parser.)

Reply:
Pytorch has a divergent behavior from numpy here where `dim=()` or `dim=[]` reduces over all dimensions. dim can also be 0 and reduce over 0th dimension so false-ish test will be wrong